lib/Sema/TypeCheckType.cpp

```diff
@@ -3558,6 +3558,10 @@ bool TypeChecker::isRepresentableInObjC(const VarDecl *VD, ObjCReason Reason) {
     return Result;
 
   SourceRange TypeRange = VD->getTypeSourceRangeForDiagnostics();
+  // TypeRange can be invalid; e.g. '@objc let foo = SwiftType()'
+  if (TypeRange.isInvalid())
+    TypeRange = VD->getNameLoc();
```

Comment:
Nitpick: *highlighting* the range isn't really correct. We should instead be able to pick a single location to pass to `diagnoseTypeNotRepresentableInObjC`.